### PR TITLE
Bug 1647451 - Follow-up improvements to Push Health details panel

### PR DESCRIPTION
**Pull request by camd**
This has a few of the tweaks we talked about in the last PR for this bug:

* I added the `max-width` overall
* Added a `developerMode` for the `DetailsPanel`/`FailureSummaryTab`.  
* I removed the red bar along the left side
* Eliminated some of the specific `CSS` classes we've brought over from the AngularJS days.
* Increased the font size in the `FailureSummaryTab` when in `developerMode`
* Removed the "file bug", "clipboard" and "log viewer" icon buttons and just made the error text a link, as we discussed.

### CODE REVIEW

File: ui/shared/tabs/failureSummary/SuggestionsListItem.jsx [hunk 1]
```diff
@@ -110,42 +111,61 @@ export default class SuggestionsListItem extends React.Component {
...
+            </span>
+          )}
         </div>
         {suggestions.length > 0 && (
           <div className="failure-summary-bugs-container">
-            <h4 className="failure-summary-bugs-title">
-              These bugs may be related:
-            </h4>
+            {developerMode && (
+              <strong>
```
Comment (sarah-clements) [inline ui/shared/tabs/failureSummary/SuggestionsListItem.jsx:165]: This seems a little overly bold. We could start using these so we get a bit closer to the mockup: https://getbootstrap.com/docs/4.0/utilities/text/#font-weight-and-italics

Comment (sarah-clements) [reply ui/shared/tabs/failureSummary/SuggestionsListItem.jsx:165]: Or add a specific font-weight, ie `font-weight: 500`

**Comment by sarah-clements** [comment]
Can you make this look closer to the mockup? :) So more spacing between the task buttons, making the bug suggestion links and bug grey (only the failure lines blue), adding the tier (if available). All of the text seems very bolded so I'm hoping we can capture some of the finer details from the mockup :) I

I'm curious about the test suite/platform name. It looks like for some there is redundant info. Could some of it be removed?

**Comment by camd** [comment]
> Can you make this look closer to the mockup? :) So more spacing between the task buttons, making the bug suggestion links and button grey (only the failure lines blue), adding the tier (if available). All of the text seems very bolded so I'm hoping we can capture some of the finer details from the mockup :)

Sure, that's fine.  I had thought our understanding was that the mockup is not considered pixel-perfect.  As such, this PR is primarily focused on the layout.  But I don't mind changing the boldness, spacing and link colors in the areas you mentioned to match the mockup.

> 
> I'm curious about the test suite/platform name. It looks like there is redundant info. Could some of it be removed? I think we did that for IFV.

Yeah, it looks like the `job_type.name` contains the platform and config in the tests I checked.  So I'll omit that.

**Comment by sarah-clements** [comment]
> > Can you make this look closer to the mockup? :) So more spacing between the task buttons, making the bug suggestion links and button grey (only the failure lines blue), adding the tier (if available). All of the text seems very bolded so I'm hoping we can capture some of the finer details from the mockup :)
> 
> Sure, that's fine. I had thought our understanding was that the mockup is not considered pixel-perfect. As such, this PR is primarily focused on the layout. But I don't mind changing the boldness, spacing and link colors in the areas you mentioned to match the mockup.
> 

Ah ok, I didn't realize you were focusing just on the layout in this pr. My understanding is that we should try to implement the design changes to match the mockup as much as possible (using an estimate since we don't have exact font sizes or weights established). I think if we use it more as a guideline we risk loosing some of the polished look and changing what areas are emphasized. Could be something to ask Victoria about.

**Comment by camd** [comment]
I went ahead and did a lot of the formatting to bring things in line with the mockup.  I opted for a softer black than the black-on-white for some parts.  I'm curious about your opinion on this.  And some of the bolding in the mockup seemed a bit too much, so I used the `font-weight: 500` as you suggested.  And I left the vertical line next to the bug suggestions that shows in Treeherder.

I went about as far as I could go on some items using Bootstrap utilities without going too far into custom `CSS`.  But if you'd prefer I match the mockup exactly, I'll do it.

**Comment by sarah-clements** [comment]
> I went ahead and did a lot of the formatting to bring things in line with the mockup. I opted for a softer black than the black-on-white for some parts. I'm curious about your opinion on this. And some of the bolding in the mockup seemed a bit too much, so I used the `font-weight: 500` as you suggested. And I left the vertical line next to the bug suggestions that shows in Treeherder.
> 
> I went about as far as I could go on some items using Bootstrap utilities without going too far into custom `CSS`. But if you'd prefer I match the mockup exactly, I'll do it.

The changes look really good :) One question about the task buttons. I see a few where they are grey. Are they grey because they are still in progress and that's why no logs are available? I'm wondering if it'd help to have the "log parsing pending" show there.

Screen Shot 2020-07-02 at 3 16 43 PM

**Comment by sarah-clements** [comment]
oh, something odd I just noticed is that if you hover over a red task button the text goes white. Is that a reactstrap bug?

**Comment by camd** [comment]
> The changes look really good :) One question about the task buttons. I see a few where they are grey. Are they grey because they are still in progress and that's why no logs are available? I'm wondering if it'd help to have the "log parsing pending" show there (in the failure summary tab instead of "failure summary is empty").
> 

Ahh, yes.  Those would be in-progress.  Yeah, I think I could add some text like that.

File: ui/shared/tabs/failureSummary/SuggestionsListItem.jsx [hunk 2]
```diff
@@ -122,7 +129,12 @@ export default class SuggestionsListItem extends React.Component {
               rel="noopener noreferrer"
               title="Go to this line in the log viewer"
             >
-              <span className="align-middle">{suggestion.search} </span>
+              <span
+                className="align-middle link-style"
+                style={{ fontWeight: '500' }}
```
Comment (sarah-clements) [inline ui/shared/tabs/failureSummary/SuggestionsListItem.jsx:134]: If this is something we're gonna use often we could just make a custom class for it :)

Comment (camd) [reply ui/shared/tabs/failureSummary/SuggestionsListItem.jsx:134]: > oh, something odd I just noticed is that if you hover over a red task button the text goes white. Is that a reactstrap bug?

It's an artifact of the conditional outline based on `isSelected`.  So I added a custom `:hover` class to handle it.

**Comment by camd** [comment]
I changed this to:

Screenshot 2020-07-02 15 59 37

File: ui/shared/tabs/failureSummary/SuggestionsListItem.jsx (hunk 2 above)
Comment (sarah-clements) [reply ui/shared/tabs/failureSummary/SuggestionsListItem.jsx]: ah ok

**Pull request merged**